Library/Source/NativeEngine.h

```diff
@@ -22,35 +22,28 @@ namespace babylon
     {
     public:
         RecycleSet(T firstId)
-            : m_nextId{ firstId }
+            : m_firstId{ firstId }
+            , m_nextId{ firstId }
         {}
 
         RecycleSet() : RecycleSet({ 0 })
         {}
 
         T Get()
         {
-            if (m_queue.empty())
-            {
-                return m_nextId++;
-            }
-            else
-            {
-                T next = m_queue.front();
-                m_queue.pop();
-                return next;
-            }
+            m_nextId ++;
+            assert(m_nextId < bgfx::getCaps()->limits.maxViews);
```

Comment:
I do not believe this change belongs in the `RecyleSet`. `RecycleSet` is a generic structure that knows nothing about bgfx or ViewIDs; it just keeps a unique and recyclable set of numbers.